spot_driver_cpp/include/spot_driver_cpp/interfaces/parameter_interface_base.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#pragma once
+
+#include <optional>
+#include <string>
+
+namespace spot_ros2
+{
+/**
+ * @brief Defines an interface for a class that retrieves the user-configured parameters needed to connect to Spot.
+ */
+class ParameterInterfaceBase
+{
+public:
+  virtual ~ParameterInterfaceBase() {};
+
+  // These functions retrieve optional parameters, where a default value can be used if the user does not provide a specific value.
+  // If the parameter was set, return the value provided by the user.
+  // If the parameter was not set, return the default value.
+  virtual std::string getAddress() const = 0;
+  virtual std::string getUsername() const = 0;
+  virtual std::string getPassword() const = 0;
+  virtual double getRGBImageQuality() const = 0;
+  virtual bool getHasRGBCameras() const = 0;
+  virtual bool getPublishRGBImages() const = 0;
+  virtual bool getPublishDepthImages() const = 0;
+  virtual bool getPublishDepthRegisteredImages() const = 0;
+  virtual std::string getSpotName() const = 0;
+
+protected:
+  // These are the definitions of the default values for optional parameters.
+  static constexpr auto kDefaultAddress = "10.0.0.3";
+  static constexpr auto kDefaultUsername = "user";
+  static constexpr auto kDefaultPassword = "password";
+  static constexpr double kDefaultRGBImageQuality{ 70.0 };
+  static constexpr bool kDefaultHasRGBCameras{ true };
+  static constexpr bool kDefaultPublishRGBImages{ true };
+  static constexpr bool kDefaultPublishDepthImages{ true };
+  static constexpr bool kDefaultPublishDepthRegisteredImages{ true };
```

Comment:
Depends on if we want this to function identically to the Python spot_ros2 driver or have it work differently. Originally I had these as non-optional, where initialization would fail if the user did not provide a value for the address, username, or password, but I changed it to how it is currently to match the multi-level fallback strategy the Python driver uses.

For the time being I think it makes sense to tend towards feature parity, and we can come back in and change it in a future PR.